minion/frontend/views.py

```diff
@@ -414,6 +422,72 @@ def persona_login():
     session['role'] = user['role']
     return api_session()
 
+def ldap_login(request):
+    data = json.loads(request.data)
+    username = data['user']
+    password = data['password']
+
+    # Initialize LDAP
+    try:
+        ld = ldap.initialize(config['login']['ldap']['uri'])
+    except ldap.LDAPError, e:
+        return jsonify(success=False, reason="error")
+
+    auth = config['login']['ldap']['uid_filter'] + "=%s," % username + config['login']['ldap']['base']
```

Comment:
Forcing a base dn will break environments that allow "baseDN-less" binds, such as Active Directory.  In AD, you simply bind as samAccountName=Wawki, and you don't need to supply a baseDN (which allows users to be anywhere in the directory).